### ISSUE TITLE
Is there any idea to reduce repetitive code?

### ISSUE THREAD

**Issue by kyrosle** [bytestring-net/bevy_lunex#81]
I feel that I have written a lot of repetitive code in the past few days. Can these cleverly take advantage of the features of bevy, such as setting UiLink and UiLayout to Bundle, or adding custom Commands to reduce repetitive code?

**Comment by IDEDARY** [comment]
Now that required components are merged, it should relieve that burden somewhat. I hope that in the future once BSN lands it will be much better.

Currently my idea is that we could abstract large ui trees into observers builders. For example a simple button is made out of several children. So we could create `Button` component and then add global observers listening to `Trigger<OnAdd, Button>` which would then spawn the pre-set nodes. This would also be done much better if we had relations, so we could keep the variables synchronized.

**Comment by davids91** [comment]
Honestly the best way for this would be to make a GUI editor, which generates the required code for the structure.

**Comment by leomeinel** [comment]
> Honestly the best way for this would be to make a GUI editor, which generates the required code for the structure.

I don't think this would be a good solution to the problem. Generating rust code from a gui tool seems like a very bad idea to me for this use case and it does not avoid duplicated code at all which still affects maintainability even if it is generated from a gui tool.

Just wanted to leave this here and it is of course only my own opinion.
